source/plugin/configuration/nodes.rst

```diff
@@ -2,47 +2,184 @@
+recommended that you always keep a reference to that root node stored somewhere.
+
+.. note::
+
+    Depending on the ``ConfigurationLoader`` used, you might even get a ``CommentedConfigurationNode``, which in
+    addition to normal ``ConfigurationNode`` behavior is able to retain a comment that will be persisted on the saved
+    config file.
+
+
+Navigating Nodes
+================
+
+Every child node is identified by a key, most commonly a string. When going from one node (for example the root node)
+to a specific child node, it may be necessary to move through multiple layers of child nodes. The keys of the child
+nodes passed make up the **path** of the target node. In the following example, there is a node holding the value
+``false`` at the path ``modules`` ``blockCheats`` ``enabled`` (from the root node) in the following HOCON representation
+of a config.
+
+.. note::
+
+    When written down, paths are commonly represented by joining the keys together with a separator char, usually ``.``.
+    The above mentioned path would be written ``modules.blockCheats.enabled``.
+
+.. code-block:: none
+
+    modules {
+        blockCheats {
+            enabled=true
+        }
     }
 
-Previously, it was mentioned that loaders can also load blank configuration nodes for you to write default data on. This
-can be done with a simple call to ``ConfigurationLoader.createEmptyNode(ConfigurationOptions)``. (and you don't even
-have an ``IOException`` to catch!)
+In Java, you can get the child node for a path using the ``getNode(...)`` method. The method accepts any number of
+arguments, where each argument is one key in the path. The path ``modules.blockCheats.enabled`` from a root node is
+acquired as follows.
 
 .. code-block:: java
 
-    File potentialFile = new File("config.conf");
-    ConfigurationLoader<CommentedConfigurationNode> loader = HoconConfigurationLoader.builder().setFile(potentialFile).build();
-    ConfigurationNode rootNode = loader.createEmptyNode(ConfigurationOptions.defaults());
+    import ninja.leaping.configurate.ConfigurationNode;
+
+    final ConfigurationNode rootNode = ...;
+    final ConfigurationNode targetNode = rootNode.getNode("modules", "blockCheats", "enabled");
+
+.. note::
+
+    The function calls ``node.getNode("foo", "bar")`` and ``node.getNode("foo").getNode("bar")`` are equivalent.
+
+Values
+======
+
+Basic Values
+~~~~~~~~~~~~
+
+Basic value types like ``int``, ``double``, ``boolean`` or ``String`` each have their own convenience getter method
+which will return the value or a default if the node does not contain a value of that type. Let's check if the server
+administrator wants our plugin to enable its blockCheats module by checking the value at the
+``modules.blockCheats.enabled`` path.
+
+.. code-block:: java
+
+    final boolean shouldEnable = rootNode.getNode("modules", "blockCheats", "enabled").getBoolean();
```

Comment:
Sometimes you use final in the examples here, sometimes you don't. Not sure if it's really important for the example.